ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,107 @@
+to be pushed to.
+
+
+**Goals**: 
+* Provide an out-of-the-box location for users’ images to be pushed after being built from source.
+
+
+**Non-Goals**: 
+* Define the user experience for users bringing existing images from other image registry services.
+* Provide users an option to choose what the out-of-the-box location for images would be. 
+* Define the user experience for users who are willing to configure additional credentials for pushing to an image registry of their choice.
+
+**Design Goals**
+* Use Quay.io as the out-of-the-box image registry in order to avoid getting into the business of maintaining an internal registry.
+* Align user permissions in an AppStudio workspace with those in Quay.io
+* Maintain the right levels of isolation between images being expected of a multi-tenant system.
+* Security:
+  * Leverage short-lived Quay.io API tokens for repo/org management on Quay.io
+  * Leverage robot accounts for pushing images from CI/Build pipelines.
+
+
+
+## Decision
+
+### What
+
+* Per workspace, setup a new org “quay.io/unique-org-david/…”
+* Per component, setup a new new repo “quay.io/unique-org-david/unique-component-repo”
+* Use User’s Quay.io API token to manage the org/repo. Short-term, we'll use a pre-configured Quay.io API token associated with AppStudio to create the org/repo till we 
+ figure out how to determinstically map a user in AppStudio to a user in Quay.io.
+* Generate a robot account token scoped to the relevant repository and persist it in the user's workspace for the image build and push process to consume.
+
+
+### How
+
+#### Long-term
+* (One-time) Use an OAuth flow to get David’s Quay.io API token if not previously pulled
+* Create a new Quay.io org owned by David using David’s API token “quay.io/unique-org-david”
+* Create a new image repo for “foo” using David’s API token “quay.io/unique-org-david/appname-foo”
+* Optionally, if needed, make the image repo public using the David’s API token for easy image pulls.
+* Configure the robot account “appstudio” in “quay.io/unique-org-david” to be able to push to “quay.io/unique-org-david/appname-foo” Invalidate/Delete/Prune David’s API token. 
+
+
+#### Short-term
+
+##### Setup
+* Configure the a pre-configured Quay.io API token for repo/org management, also known as the OAuth Access token. From a security perspective, this would be treated as a typical service account token
+  belonging to AppStudio. This token would be shortlived and rotated *as frequently as practically possible*  .
+
+##### When a user creates a Component
+1. Create a new Quay.io org owned by David using the pre-configured AppStudio-wide API token `quay.io/unique-org-david`
+2. Create a new image repo for Component “foo” using the the pre-configured David’s API token `quay.io/unique-org-david/unique-component-foo`
+3. Optionally, if needed, make the image repo public using the API token for easy image pulls.
```

Comment:
We can not make the repository public. The user owns the repo, it should be their decision to make it public. Otherwise, we will risk exposing users' private applications. Other than David nobody else should be able to access it.

Reply:
sounds good 👍 "private" is anyway the 'default' in quay.io . For private repositories, we'd have to propagate the pull secret to the deployment environment.